rent/src/lib.rs

```diff
@@ -0,0 +1,1611 @@
+mod types;
+mod weights;
+
+pub use pallet::*;
+pub use types::*;
+
+use frame_support::{
+	dispatch::{DispatchError, DispatchResult},
+	ensure,
+	pallet_prelude::DispatchResultWithPostInfo,
+	traits::{
+		Currency,
+		ExistenceRequirement::{AllowDeath, KeepAlive},
+		Get, StorageVersion,
+	},
+	BoundedVec, PalletId,
+};
+use frame_system::{pallet_prelude::*, RawOrigin};
+use sp_runtime::{
+	traits::{AccountIdConversion, CheckedDiv, CheckedSub, Saturating},
+	SaturatedConversion,
+};
+use sp_std::prelude::*;
+
+use primitives::nfts::{NFTData, NFTId};
+use ternoa_common::traits::NFTExt;
+pub use weights::WeightInfo;
+
+pub type BalanceOf<T> =
+	<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+pub type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<
+	<T as frame_system::Config>::AccountId,
+>>::NegativeImbalance;
+
+const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	#[pallet::storage_version(STORAGE_VERSION)]
+	pub struct Pallet<T>(PhantomData<T>);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Weight information for pallet.
+		type WeightInfo: WeightInfo;
+
+		/// Currency type.
+		type Currency: Currency<Self::AccountId>;
+
+		/// Link to the NFT pallet.
+		type NFTExt: NFTExt<AccountId = Self::AccountId>;
+
+		// Constants
+		/// The auctions pallet id - will be used to generate account id.
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+
+		/// The maximum number of accounts that can be stored inside the account list.
+		#[pallet::constant]
+		type AccountSizeLimit: Get<u32>;
+
+		/// Maximum number of simultaneous rent contract.
+		#[pallet::constant]
+		type SimultaneousContractLimit: Get<u32>;
+
+		/// Maximum number of related automatic rent actions in block.
+		#[pallet::constant]
+		type ActionsInBlockLimit: Get<u32>;
+
+		/// Maximum number of blocks during which a rent contract is available for acceptance.
+		#[pallet::constant]
+		type ContractExpirationDuration: Get<u32>;
+	}
+
+	/// Data related to rent contracts.
+	#[pallet::storage]
+	#[pallet::getter(fn contracts)]
+	pub type Contracts<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		NFTId,
+		RentContractData<T::AccountId, T::BlockNumber, BalanceOf<T>, T::AccountSizeLimit>,
+		OptionQuery,
+	>;
+
+	#[pallet::type_value]
+	pub fn DefaultNumberOfCurrentContracts() -> u32 {
+		0u32
+	}
+
+	/// Data related to fixed contract deadlines.
+	#[pallet::storage]
+	#[pallet::getter(fn number_of_current_contracts)]
+	pub type NumberOfCurrentContracts<T: Config> = StorageValue<_, u32, ValueQuery, DefaultNumberOfCurrentContracts>;
```

Comment:
You don't need `DefaultNumberOfCurrentContracts` since the default value is `0`,